src/app.ts

```diff
@@ -236,11 +237,38 @@ export class App {
    *   ctx.res.set('X-Mojo', 'I <3 mojo.js!');
    *   await ctx.render(...args);
    * });
+   *
+   * // Render response with header using nested helper
+   * app.addHelper('renderWith.header', async (ctx, ...args) => {
+   *   ctx.res.set('X-Mojo', 'I <3 mojo.js!');
+   *   await ctx.render(...args);
+   * });
    */
   addHelper(name: string, fn: MojoAction): this {
-    return this.decorateContext(name, function (this: MojoContext, ...args: any[]) {
-      return fn(this, ...args);
-    });
+    const nestedNames = name.split('.');
+    if (nestedNames.length == 1) {
+      return this.decorateContext(name, function (this: MojoContext, ...args: any[]) {
+        return fn(this, ...args);
+      });
+    } else if (nestedNames.length == 2) {
+      const [getterName, methodName] = nestedNames;
+      this._nestedFunctions[getterName] ??= {};
+      this._nestedFunctions[getterName][methodName] = fn;
+      const fnEntries = Object.entries<MojoAction>(this._nestedFunctions[getterName]);
+      return this.decorateContext(getterName, {
+        get: function () {
+          const ctxScopedFunctions: any = {};
+          for (const [key, fn] of fnEntries)
```

Comment:
Please no block-less loops and conditions. They are so ugly.

Reply:
done